assets/src/dashboard/components/typeahead-input/index.js

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useCallback, useMemo, useRef, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { ReactComponent as SearchIcon } from '../../icons/search.svg';
+import { ReactComponent as CloseIcon } from '../../icons/close.svg';
+import useFocusOut from '../../utils/useFocusOut';
+import { TypeaheadOptions } from '../';
+
+const SearchContainer = styled.div`
+  width: 272px;
+  position: static;
+  display: flex;
+  flex-direction: column;
+  border-radius: ${({ theme, isOpen }) =>
+    isOpen ? theme.border.expandedTypeaheadRadius : 'none'};
+  border: none;
+  box-shadow: ${({ theme, isOpen }) =>
+    isOpen ? theme.boxShadow.expandedTypeahead : 'none'};
+`;
+SearchContainer.propTypes = {
+  isOpen: PropTypes.bool,
+};
+
+const InputContainer = styled.div`
+  position: relative;
+  display: flex;
+  flex-direction: row;
+  width: 100%;
+  height: 48px;
+  padding-top: 15px;
+  padding-bottom: 15px;
+  padding-left: 19px;
+  border-radius: ${({ theme, isOpen }) =>
+    isOpen ? 'none' : theme.border.typeaheadRadius};
+  border: none;
+  border-bottom: ${({ theme, isOpen }) =>
+    isOpen && `1px solid ${theme.colors.gray50}`};
+  color: ${({ theme }) => theme.colors.gray500};
+  background-color: ${({ theme, isOpen }) =>
+    isOpen ? theme.colors.white : theme.colors.gray25};
+`;
+InputContainer.propTypes = {
+  isOpen: PropTypes.bool,
+};
+
+const StyledInput = styled.input`
+  align-self: center;
+  border: none;
+  background-color: transparent;
+  text-overflow: ellipsis;
+  padding: 0 11.95px;
+  margin: auto 0;
+  height: 100%;
+  font-family: ${({ theme }) => theme.fonts.typeaheadInput.family};
+  font-size: ${({ theme }) => theme.fonts.typeaheadInput.size};
+  line-height: ${({ theme }) => theme.fonts.typeaheadInput.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.typeaheadInput.letterSpacing};
+  font-weight: ${({ theme }) => theme.fonts.typeaheadInput.weight};
+  color: ${({ theme }) => theme.colors.gray500};
+
+  &:disabled {
+    cursor: default;
+  }
+`;
+
+const IconContainer = styled.span`
+  margin: auto 0;
+  width: 17.05px;
```

Comment:
There are a lot of interesting pixel values in this PR, including fractional values. Are these coming from Figma?

Reply:
yes they are all coming from figma

Reply:
Just talked to @samitron7 and she is saying that all measurements should be in a 4dp grid. Would you mind connecting with her on Slack to confirm this. Just in case there's some misunderstanding.

Reply:
I am slacking with her now and will share what I find. For reference, these are the specs I was going off of. 
![Screen Shot 2020-03-25 at 2 17 48 PM](https://user-images.githubusercontent.com/10720454/77586656-ce6a0280-6ea3-11ea-958f-b4ca2e4358ca.png)
![Screen Shot 2020-03-25 at 2 18 09 PM](https://user-images.githubusercontent.com/10720454/77586658-cf029900-6ea3-11ea-9cef-0f37e37924f8.png)
![Screen Shot 2020-03-25 at 2 18 15 PM](https://user-images.githubusercontent.com/10720454/77586660-cf9b2f80-6ea3-11ea-9f8f-c6cfc8a6b426.png)

Reply:
From Sam:

> Hey Brittany - oh man this is terrible! So this is all design agency work and you think they use whole numbers. I'm using a 4dp grid on the editor. For sizing, all the icons, input fields etc should match. Until I get a chance to reflect those changes (which might take awhile because I have a list of things to do), can you refer to the editor figma for reference and just make sure everything is divisible by 4.